app/services/follow_remote_account_service.rb

```diff
@@ -28,21 +28,30 @@ def call(uri)
 
     return Account.find_local(confirmed_username) if TagManager.instance.local_domain?(confirmed_domain)
 
-    confirmed_account = Account.find_remote(confirmed_username, confirmed_domain)
-    return confirmed_account unless confirmed_account.nil?
+    remote_url  = data.link('http://schemas.google.com/g/2010#updates-from').href
+    salmon_url  = data.link('salmon').href
+    url         = data.link('http://webfinger.net/rel/profile-page').href
+    public_key  = magic_key_to_pem(data.link('magic-public-key').href)
 
-    Rails.logger.debug "Creating new remote account for #{uri}"
+    account = Account.find_remote(confirmed_username, confirmed_domain)
+    return account unless account.nil? || remote_url != account.remote_url || salmon_url != account.salmon_url || url != account.url || public_key != account.public_key
```

Comment:
That is what I went for in this patch: re-fetch any remote account (both from webfinger and the feed) for which we have not refreshed for at least one day using a “last_webfingered_at” field.
I have not seen any ill effect on my small instance.